src/datasources/cache/cache.router.ts

```diff
@@ -403,4 +432,30 @@ export class CacheRouter {
   static getExchangeRatesCacheDir(): CacheDir {
     return new CacheDir(CacheRouter.EXCHANGE_RATES_KEY, '');
   }
+
+  static getNativeCoinPriceCacheDir(args: {
+    nativeCoinId: string;
+    fiatCode: string;
+    tokenAddress?: string;
+  }): CacheDir {
+    return new CacheDir(
+      `${args.nativeCoinId}_${CacheRouter.NATIVE_COIN_PRICE_KEY}`,
+      args.fiatCode,
+    );
+  }
+
+  static getTokenPriceCacheDir(args: {
+    chainName: string;
+    fiatCode: string;
+    tokenAddress?: string;
```

Comment:
Can you get a cached token price when `tokenAddress` is `undefined`?

Reply:
Good point. It shouldn't be possible. I'll make it mandatory here 👍

Reply:
Fixed in [d4cf6711](https://github.com/safe-global/safe-client-gateway/commit/d4cf6711d2b6f399acd83fd4be5b8ff10e1b778d)